jbmc/src/java_bytecode/java_object_factory.cpp

```diff
@@ -894,11 +894,53 @@ void java_object_factoryt::gen_nondet_struct_init(
       code_function_callt{init_func->symbol_expr(), {address_of_exprt{expr}}});
 }
 
+/// Generate code block that verifies that an expression of type float or
+/// double has integral value. For example, for a float expression floatVal we
+/// generate:
+/// int assume_integral_tmp;
+/// assume_integral_tmp = NONDET(int);
+/// ASSUME FLOAT_TYPECAST(assume_integral_tmp, float, __CPROVER_rounding_mode)
+///   == floatVal
+/// \param expr The expression we want to make an assumption on
+/// \param type The type of the expression
+/// \param location Source location associated with the expression
+/// \param [out] symbol_table Symbol table, a new auxiliary symbol will be added
+/// \return Code block constructing the auxiliary nondet integer and an
+///   assume statement that the integer is equal to the value of the expression
+static code_blockt assume_expr_integral(
+  const exprt &expr,
+  const typet &type,
+  const source_locationt &location,
+  symbol_table_baset &symbol_table)
+{
+  PRECONDITION(type == java_float_type() || type == java_double_type());
+
+  code_blockt assignments;
+  const symbolt &aux_symbol = fresh_java_symbol(
+    java_int_type(),
+    "assume_integral_tmp",
+    location,
+    "assume_integral_tmp",
+    symbol_table);
+  const auto &aux_expr = aux_symbol.symbol_expr();
+  assignments.add(code_declt(aux_expr), location);
+  exprt aux_rhs = side_effect_expr_nondett(java_int_type(), location);
```

Comment:
:bulb: Might want to call this `nondet` rather than `aux` as a bit more informative

Reply:
Done